### DISCUSSION TITLE
Cloud firestore says Document not found

### DISCUSSION THREAD

**Discussion by g-mahendra** [invertase/react-native-firebase#4805] [Q&A] [answered]
I am building an app where I create a document  with the name as some random string generated and is succeeds  Code bellow.

`import React, {useState} from 'react';
import {View, Text, StyleSheet, TextInput} from 'react-native';
import {Button} from 'react-native-paper';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';
import {useAuth} from './context/AuthContext';
import firestore from '@react-native-firebase/firestore';

const CreateGame = ({route, navigation}) => {
  const [host, setHost] = useState('');
  const {currentUser} = useAuth();

  const {code} = route.params;
  return (
    <View style={styles.container}>
      <Text style={styles.title}>
        Share bellow code with your friend for him/her to join the game
      </Text>
      <View style={{...styles.btn, backgroundColor: '#120040'}}>
        <Text style={styles.gameCode}>{code}</Text>
      </View>
      <View style={styles.input}>
        <TextInput
          value={host}
          onChangeText={(text) => setHost(text)}
          placeholder="Enter username for game"
          placeholderTextColor="#aaa"
          color="#FFF"
        />
      </View>
      <Button
        onPress={() => {
          firestore()
            .collection('native')
            .doc(`${code}`)
            .set({
              gamecode: code,
              host: host,
              hostEmail: currentUser.email,
              guest: '',
              hostmove: '',
              guestmove: '',
              winner: ''
            })
            .then(() => console.log('Game created successfully'))
            .catch((e) => console.log(e));
          navigation.navigate('game', {
            code: code,
          });
        }}
        style={styles.btn}>
        Enter Game
      </Button>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
    backgroundColor: '#12005e',
  },
  btn: {
    backgroundColor: '#FFF',
    borderRadius: wp('10%'),
    width: wp('50%'),
    paddingVertical: hp('2%'),
    marginVertical: hp('2%'),
    alignItems: 'center',
  },
  title: {
    fontSize: hp('2%'),
    color: '#FFF',
    marginVertical: hp('2%'),
  },
  gameCode: {
    fontSize: hp('3%'),
    color: '#FFF',
  },
  input: {
    borderRadius: wp('10%'),
    width: wp('50%'),
    paddingVertical: hp('2%'),
    marginVertical: hp('2%'),
    alignItems: 'center',
    backgroundColor: '#120040',
    paddingVertical: hp('1%'),
    borderWidth: 1,
    borderColor: '#FFF',
  },
});

export default CreateGame;
`

But when I try to update some value of that same document, ( that means I know the name of document in advance ) it shows an error like  bellow
`[Error: [firestore/not-found] Some requested document was not found.]
`
`The failing code is attached bellow. Thanks in advance.
import React, {useState} from 'react';
import {View, Text, StyleSheet, TextInput} from 'react-native';
import {Button} from 'react-native-paper';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';
import {useAuth} from './context/AuthContext';
import firestore from '@react-native-firebase/firestore';

const JoinGame = ({route, navigation}) => {
  const [guest, setGuest] = useState('');
  const {currentUser} = useAuth();
  const [code, setCode] = useState('');

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Enter the game code to join the game</Text>
      <View style={styles.input}>
        <TextInput
          value={guest}
          onChangeText={(text) => setGuest(text)}
          placeholder="Enter username for game"
          placeholderTextColor="#aaa"
          color="#FFF"
        />
      </View>
      <View style={styles.input}>
        <TextInput
          value={code}
          onChangeText={(text) => setCode(text)}
          placeholder="Enter game code"
          placeholderTextColor="#aaa"
          color="#FFF"
        />
      </View>
      <Button
        onPress={() => {
          firestore()
            .collection('native')
            .doc(`${code}`)
            .update({
              guest: guest,
            })
            .then(() => console.log('Updated'))
            .catch((e) => console.log(e));
          // navigation.navigate('game', {
          //   code: code,
          // });
        }}
        style={styles.btn}>
        Join Game
      </Button>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
    backgroundColor: '#12005e',
  },
  btn: {
    backgroundColor: '#FFF',
    borderRadius: wp('10%'),
    width: wp('50%'),
    paddingVertical: hp('2%'),
    marginVertical: hp('2%'),
    alignItems: 'center',
  },
  title: {
    fontSize: hp('2%'),
    color: '#FFF',
    marginVertical: hp('2%'),
  },
  gameCode: {
    fontSize: hp('3%'),
    color: '#FFF',
  },
  input: {
    borderRadius: wp('10%'),
    width: wp('50%'),
    paddingVertical: hp('2%'),
    marginVertical: hp('2%'),
    alignItems: 'center',
    backgroundColor: '#120040',
    paddingVertical: hp('1%'),
    borderWidth: 1,
    borderColor: '#FFF',
  },
});

export default JoinGame;
`

**Comment by mikehardy** [comment, marked as answer]
This is not the way to go about filing a bug report in this repository, sorry. If you have an issue, please open an issue following the template. Please make sure you have up to date versions, and provide a self-contained App.js that reproduces the problem (including any necessary data creation) when dropped in a clean project. 99% of the time, the act of reducing the problem to a reproducible example on a clean project shows a coding error and that's the answer - when it doesn't, we have a clean basis to reproduce it and issue a fix. Thanks

**Reply by g-mahendra** [reply]
I updated all dependencies to latest version and it worked. Thanks!
